components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/internal/dao/PermissionsDAO.java

```diff
@@ -243,6 +244,46 @@ public boolean hasPermission(List<Role> roles, Permission permission) {
         return hasPermission;
     }
 
+    /**
+     * Get granted roles of a specific permission.
+     *
+     * @param permission
+     * @return
+     */
+    public List<Role> getGrantedRoles(Permission permission) {
+        List<Role> roles = Collections.emptyList();
+        Connection conn = null;
+        PreparedStatement ps = null;
+        ResultSet resultSet = null;
+        String query = "SELECT * FROM ROLE_PERMISSIONS WHERE APP_NAME = ? AND PERMISSION_STRING = ?";
```

Comment:
Since we only need the ROLE_ID can't we change this query to "SELECT ROLE_ID FROM ROLE_PERMISSIONS WHERE APP_NAME = ? AND PERMISSION_STRING = ?"

Reply:
Yes we can. Will do the change.